### PR TITLE
Checkmarx sast fixes

### PR DESCRIPTION
**Pull request by brohini**
[//]: * (If not ready for merging, indicate the reason after the # on the next line.)
#
#### Card(s)
[//]: * (Link to the PivotalTracker or LeanKit cards.)
- [Fix checkmarx SAST issues (**HIGH**) for OpenVXML](https://www.pivotaltracker.com/story/show/183287678)

#### Changes Made
1. Fixed checkmarx SAST high vulnerabilities (SQL injection errors)

### CODE REVIEW

**Comment by glyphrider** [comment]
Logo
**Checkmarx AST – Scan Summary & Details** – [1053c84c\-1171\-4f11\-9cc8\-7502013a110e](https://ast.checkmarx.net/projects/69538ccd-a742-4a8d-aff9-873706781845/scans?id=1053c84c-1171-4f11-9cc8-7502013a110e&branch=checkmarx_SAST_fixes)
### New Issues
Severity|Issue|File / Package|Scan Engine
---|---|---|---
HIGH|CVE\-2022\-42003|Maven\-com\.fasterxml\.jackson\.core:jackson\-databind\-2\.13\.3|[CxSCA](https://ast.checkmarx.net/sca/#/projects/69538ccd-a742-4a8d-aff9-873706781845/reports/1053c84c-1171-4f11-9cc8-7502013a110e/vulnerabilities/all)
HIGH|CVE\-2022\-42004|Maven\-com\.fasterxml\.jackson\.core:jackson\-databind\-2\.13\.3|[CxSCA](https://ast.checkmarx.net/sca/#/projects/69538ccd-a742-4a8d-aff9-873706781845/reports/1053c84c-1171-4f11-9cc8-7502013a110e/vulnerabilities/all)
MEDIUM|Unchecked\_Input\_for\_Loop\_Condition|/com\.vht\.openvxml\.desktop/com\.vht\.openvxml\.desktop\.plugins/org\.eclipse\.vtp\.desktop\.projects\.interactive\.core/src/main/java/org/eclipse/vtp/desktop/projects/interactive/core/wizards/CreateInteractiveWorkflowProjectWizard\.java: [165](https://github.com/VHT/OpenVXML/blob/checkmarx_SAST_fixes//com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/org.eclipse.vtp.desktop.projects.interactive.core/src/main/java/org/eclipse/vtp/desktop/projects/interactive/core/wizards/CreateInteractiveWorkflowProjectWizard.java#L165)|[CxSAST](https://ast.checkmarx.net/results/1053c84c-1171-4f11-9cc8-7502013a110e/69538ccd-a742-4a8d-aff9-873706781845/sast)
MEDIUM|Unchecked\_Input\_for\_Loop\_Condition|/com\.vht\.openvxml\.desktop/com\.vht\.openvxml\.desktop\.plugins/org\.eclipse\.vtp\.desktop\.projects\.core/src/main/java/org/eclipse/vtp/desktop/projects/core/wizards/CreateDatabaseTableWizard\.java: [124](https://github.com/VHT/OpenVXML/blob/checkmarx_SAST_fixes//com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/org.eclipse.vtp.desktop.projects.core/src/main/java/org/eclipse/vtp/desktop/projects/core/wizards/CreateDatabaseTableWizard.java#L124)|[CxSAST](https://ast.checkmarx.net/results/1053c84c-1171-4f11-9cc8-7502013a110e/69538ccd-a742-4a8d-aff9-873706781845/sast)
MEDIUM|Unchecked\_Input\_for\_Loop\_Condition|/com\.vht\.openvxml\.desktop/com\.vht\.openvxml\.desktop\.plugins/org\.eclipse\.vtp\.desktop\.projects\.core/src/main/java/org/eclipse/vtp/desktop/projects/core/wizards/CreateWorkflowProjectWizard\.java: [131](https://github.com/VHT/OpenVXML/blob/checkmarx_SAST_fixes//com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/org.eclipse.vtp.desktop.projects.core/src/main/java/org/eclipse/vtp/desktop/projects/core/wizards/CreateWorkflowProjectWizard.java#L131)|[CxSAST](https://ast.checkmarx.net/results/1053c84c-1171-4f11-9cc8-7502013a110e/69538ccd-a742-4a8d-aff9-873706781845/sast)
MEDIUM|Unchecked\_Input\_for\_Loop\_Condition|/com\.vht\.openvxml\.desktop/com\.vht\.openvxml\.desktop\.plugins/org\.eclipse\.vtp\.desktop\.projects\.interactive\.core/src/main/java/org/eclipse/vtp/desktop/projects/interactive/core/wizards/CreateUmbrellaProjectWizard\.java: [151](https://github.com/VHT/OpenVXML/blob/checkmarx_SAST_fixes//com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/org.eclipse.vtp.desktop.projects.interactive.core/src/main/java/org/eclipse/vtp/desktop/projects/interactive/core/wizards/CreateUmbrellaProjectWizard.java#L151)|[CxSAST](https://ast.checkmarx.net/results/1053c84c-1171-4f11-9cc8-7502013a110e/69538ccd-a742-4a8d-aff9-873706781845/sast)
MEDIUM|Unchecked\_Input\_for\_Loop\_Condition|/com\.vht\.openvxml\.desktop/com\.vht\.openvxml\.desktop\.plugins/org\.eclipse\.vtp\.desktop\.projects\.voice/src/main/java/org/eclipse/vtp/desktop/projects/voice/wizards/CreateVoiceWizard\.java: [96](https://github.com/VHT/OpenVXML/blob/checkmarx_SAST_fixes//com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/org.eclipse.vtp.desktop.projects.voice/src/main/java/org/eclipse/vtp/desktop/projects/voice/wizards/CreateVoiceWizard.java#L96)|[CxSAST](https://ast.checkmarx.net/results/1053c84c-1171-4f11-9cc8-7502013a110e/69538ccd-a742-4a8d-aff9-873706781845/sast)

### Fixed Issues
Severity|Issue|File / Package|Scan Engine
---|---|---|---
HIGH|Second\_Order\_SQL\_Injection|/com\.vht\.openvxml\.desktop/com\.vht\.openvxml\.desktop\.plugins/com\.openmethods\.openvxml\.desktop\.model\.workflow/src/com/openmethods/openvxml/desktop/model/workflow/internal/WorkflowIndex\.java: [733](https://github.com/VHT/OpenVXML/blob/checkmarx_SAST_fixes//com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/com.openmethods.openvxml.desktop.model.workflow/src/com/openmethods/openvxml/desktop/model/workflow/internal/WorkflowIndex.java#L733), [1217](https://github.com/VHT/OpenVXML/blob/checkmarx_SAST_fixes//com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/com.openmethods.openvxml.desktop.model.workflow/src/com/openmethods/openvxml/desktop/model/workflow/internal/WorkflowIndex.java#L1217), [1276](https://github.com/VHT/OpenVXML/blob/checkmarx_SAST_fixes//com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/com.openmethods.openvxml.desktop.model.workflow/src/com/openmethods/openvxml/desktop/model/workflow/internal/WorkflowIndex.java#L1276)|[CxSAST](https://ast.checkmarx.net/results/1053c84c-1171-4f11-9cc8-7502013a110e/69538ccd-a742-4a8d-aff9-873706781845/sast)

File: com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/com.openmethods.openvxml.desktop.model.workflow/src/com/openmethods/openvxml/desktop/model/workflow/internal/WorkflowIndex.java [hunk 1]
```diff
@@ -93,10 +94,11 @@ public List<IWorkflowEntry> getWorkflowEntries(
 		try {
 			if (documentId != null) {
 				Connection con = createConnection(true);
-				Statement st = con.createStatement();
+				String query = "select id, name from workflowentries where documentid = (?)";
+				PreparedStatement st = con.prepareStatement(query);
+				st.setString(1, documentId);
```
Comment (jspencer92) [inline com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/com.openmethods.openvxml.desktop.model.workflow/src/com/openmethods/openvxml/desktop/model/workflow/internal/WorkflowIndex.java:99]: @brohini are these Id values actually expected to be strings on the DB side? In my mind, Id values are typically int data types, and may throw an error depending on what data type the DB is expecting.

**Review by jspencer92 — APPROVED** [review]
Just the one question on the Id data types. If the DB is expecting int, we would want to use setInt in place of setString. If the DB is truly expecting strings for those values, you can ignore my comment.

File: com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/com.openmethods.openvxml.desktop.model.workflow/src/com/openmethods/openvxml/desktop/model/workflow/internal/WorkflowIndex.java (hunk 1 above)
Comment (brohini) [reply com.vht.openvxml.desktop/com.vht.openvxml.desktop.plugins/com.openmethods.openvxml.desktop.model.workflow/src/com/openmethods/openvxml/desktop/model/workflow/internal/WorkflowIndex.java:99]: @jspencer92 They were Strings. I have checked the query execution in build process, no exceptions were raised.
There are some places that is expecting int, I used setInt() for those.

**Pull request merged**